### PR TITLE
Documentation improvements

### PR DESCRIPTION
**Pull request by zackgalbreath**
* Clarify documentation for dependency management with CMake
* Improve FetchContent example in docs

@jtattermusch

### CODE REVIEW

File: BUILDING.md [hunk 1]
```diff
@@ -172,13 +172,21 @@ If you want to build DLLs, run `cmake` with `-DBUILD_SHARED_LIBS=ON`.
 
 ### Dependency management
 
-gRPC's CMake build system provides two modes for handling dependencies.
+gRPC's CMake build system has two options for handling dependencies.
+CMake can build the dependencies for you, or it can search for libraries
+that are already installed on your system and use them to build gRPC.
+
+This behavior is controlled by the `gRPC_<depname>_PROVIDER` CMake variables,
+ie `gRPC_CARES_PROVIDER`. The options that these variables take are as follows:
```
Comment (jtattermusch) [inline BUILDING.md]: nit: s/ie/e.g./ ?

File: BUILDING.md [hunk 2]
```diff
@@ -172,13 +172,21 @@ If you want to build DLLs, run `cmake` with `-DBUILD_SHARED_LIBS=ON`.
...
+
 * module - build dependencies alongside gRPC.
 * package - use external copies of dependencies that are already available
```
Comment (jtattermusch) [inline BUILDING.md]: nit: add "(preferrably you'd pre-install these also using cmake, with the `CMAKE_INSTALL_PREFIX` option)" ?

File: BUILDING.md [hunk 3]
```diff
@@ -172,13 +172,21 @@ If you want to build DLLs, run `cmake` with `-DBUILD_SHARED_LIBS=ON`.
...
 
-gRPC's CMake build system provides two modes for handling dependencies.
+gRPC's CMake build system has two options for handling dependencies.
+CMake can build the dependencies for you, or it can search for libraries
+that are already installed on your system and use them to build gRPC.
+
+This behavior is controlled by the `gRPC_<depname>_PROVIDER` CMake variables,
+ie `gRPC_CARES_PROVIDER`. The options that these variables take are as follows:
+
 * module - build dependencies alongside gRPC.
```
Comment (jtattermusch) [inline BUILDING.md]: nit: add "(the source code will be obtained from the git submodules)"?

File: src/cpp/README.md [hunk 4]
```diff
@@ -70,18 +70,26 @@ also sets up an `add_subdirectory()` rule for you. This causes gRPC to be
...
 FetchContent_Declare(
   gRPC
   GIT_REPOSITORY https://github.com/grpc/grpc
   GIT_TAG        v1.25.0
 )
+set(FETCHCONTENT_QUIET OFF)
 FetchContent_MakeAvailable(gRPC)
 
-add_executable(my_exe my_exe.cc)
+add_executable(my_exe my_exe.cxx)
```
Comment (jtattermusch) [inline src/cpp/README.md]: why cxx and not .cc? (this repo uses .cc for C++ files)

File: BUILDING.md (hunk 1 above)
Comment (zackgalbreath) [reply BUILDING.md]: 👍

File: BUILDING.md (hunk 2 above)
Comment (zackgalbreath) [reply BUILDING.md]: 👍

File: BUILDING.md (hunk 3 above)
Comment (zackgalbreath) [reply BUILDING.md]: 👍

File: src/cpp/README.md (hunk 4 above)
Comment (zackgalbreath) [reply src/cpp/README.md]: 👍

**Comment by zackgalbreath** [comment]
Thanks for the feedback, @jtattermusch . I believe I've addressed your concerns. Please let me know if you'd like to see any more changes before we merge this.

**Comment by zackgalbreath** [comment]
@jtattermusch can we merge this?

**Comment by jtattermusch** [comment]
CC @chalin  FYI

**Comment by jtattermusch** [comment]
@

> @jtattermusch can we merge this?

Sure. Sorry, this feel through the cracks.

**Pull request merged**